src/interval_sets.jl

```diff
@@ -1,7 +1,28 @@
 
 ###### Set-related Helpers #####
+"""
+    IntervalSet{T<:AbstractInterval} <: AbstractSet{T}
+
+A set type for performing multi-interval specific operations.
+https://en.wikipedia.org/wiki/Interval_arithmetic#Interval_operators
+"""
+struct IntervalSet{T<:AbstractInterval} <: AbstractSet{T}
+    # TODO: Use Dict internally once union!(Vector{AbstractInterval}) is fully deprecated
```

Comment:
In relation to a point made by @omus, we need to wrap a vector without mutating or copying in order to properly deprecated the old `union!` behaviour. Once that's gone we should be good to use a more appropriate internal type like a `Dict`.